### PR TITLE
[WIP] Check PR title and label as if it's wip then ignore Danger check

### PR DESCRIPTION
**Pull request by byhbt**
## What happened
Check PR title and label as if it's wip then ignore Danger check

 
## Insight
n/a
 

## Proof Of Work

The specs should pass.

### CODE REVIEW

File: Dangerfile [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# Make it more obvious that a PR is a work in progress and shouldn't be merged yet.
+has_wip_label = github.pr_labels.any? { |label| label.include? "WIP" }
+has_wip_title = github.pr_title.include? "[WIP]" || github.pr_title.include? "WIP"
```
Comment (andyduong1920) [inline Dangerfile:3]: can we check 1 more thing @byhbt :D checking is_draft PR 😁  so we can cover all the cases :D

File: Dangerfile [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+
+if has_wip_label || has_wip_title
+  message("Skipping Danger since PR is classed as Work in Progress")
```
Comment (andyduong1920) [inline Dangerfile:6]: Small thing, how about `Skipping Danger since PR state is in Work In Progress state`

Comment (byhbt) [reply Dangerfile:6]: No, in your sentence, `state` word is duplicated 😄  Any other idea? 🤔

File: Dangerfile (hunk 1 above)
Comment (hoangmirs) [reply Dangerfile:3]: I think this case can cover the previous one
`github.pr_title.include? "WIP"`

Comment (hoangmirs) [reply Dangerfile:3]: And this maybe helps you to find the draft state
https://spectrum.chat/danger/javascript/support-for-github-draft-prs~82948576-ce84-40e7-a043-7675e5bf5690

**Comment by byhbt** [comment]
As this morning @longnd raise an issue about running Danger on a pushed branch which has not create PR yet. So we should not run Danger on that branch as well by adding the param `fail-if-no-pr=true`

Reference:
https://github.com/danger/danger/blob/master/lib/danger/commands/runner.rb

**Comment by andyduong1920** [comment]
Hi @byhbt Could you have a look on this PR again

File: Dangerfile (hunk 2 above)
Comment (andyduong1920) [reply Dangerfile:6]: Oh, how about `Skipping Danger since PR state is in Work In Progress (WIP)`

**Comment by andyduong1920** [comment]
Ping @byhbt 👋 any update on this PR :)

**Comment by byhbt** [comment]
> Ping @byhbt 👋 any update on this PR :)

No new update, Andy, other time.

**Comment by ankitkalia1195** [comment]
@byhbt  @andyduong1920 Should we close this PR as no action taken since long time.

**Comment by andyduong1920** [comment]
> @byhbt @andyduong1920 Should we close this PR as no action taken since long time.

cc @byhbt 👆

**Pull request closed without merging**